Sources/ProtobufferBuilder/ProtobufferBuilder.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  File.swift
+//
+//
+//  Created by Nityananda on 26.11.20.
+//
+
+import Runtime
+
+/// ProtobufferBuilder builds `.proto` files.
+///
+/// Call `ProtobufferBuilder.description` for the final output.
+public class ProtobufferBuilder {
+    var messages: Set<Message>
+    var services: Set<Service>
+    
+    /// Create an instance of `ProtobufferBuilder`.
+    public init() {
+        self.messages = Set()
+        self.services = Set()
+    }
+}
+
+public extension ProtobufferBuilder {
+    /// `addService` builds a Protobuffer service declaration from the type parameter.
+    /// - Parameter type: the type of the service
+    /// - Throws: `Error`s of type `Exception`
+    func addService<T>(of type: T.Type = T.self) throws {
+        guard let serviceNode = try Tree<TypeInfo>.make(type) else {
+            return
+        }
+        
+        let serviceName = try serviceNode.value.compatibleName() + "Service"
+        
+        let messageTree = try Tree<TypeInfo>.make(type)
+            .edited(fixArray)
+            .filter { typeInfo in
+                !ParticularType(typeInfo.type).isPrimitive
+            }
+            .map { typeInfo in
+                try Message(typeInfo: typeInfo)
+            }
+        
+        guard let message = messageTree?.value else {
+            return
+        }
+        
+        let voidMessage = Message.void
+        let method = Service.Method(
+            name: "handle",
+            input: voidMessage,
```

Comment:
just for my understanding, this is where the list of parameters will be used in the future and voidMessage is just temporary?

Reply:
Yes, now it's handled.